RickAndMorty-iOS/Views/Locations/LocationsViewController.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  LocationsViewController.swift
+//  RickAndMorty-iOS
+//
+//  Created by Calvin Pak on 2023-03-13.
+//
+
+import UIKit
+import Combine
+
+class LocationsViewController: UIViewController {
+
+    let locationsView = LocationsView()
+    let viewModel = LocationsViewModel()
+    weak var coordinator: MainCoordinator?
+
+    typealias DataSource = UICollectionViewDiffableDataSource<Section, AnyHashable>
+    typealias Snapshot = NSDiffableDataSourceSnapshot<Section, AnyHashable>
+    private var dataSource: DataSource!
+    private var cancellables = Set<AnyCancellable>()
+    var snapshot = Snapshot()
+
+    override func loadView() {
+        view = locationsView
+        navigationController?.navigationBar.prefersLargeTitles = true
+        title = "Locations"
+        locationsView.collectionView.delegate = self
+        locationsView.collectionView.refreshControl = UIRefreshControl()
+        locationsView.collectionView.refreshControl?.addTarget(self, action: #selector(onRefresh), for: .valueChanged)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureDataSource()
+        showEmptyData()
+        subscribeToViewModel()
+        viewModel.currentPage = 1
```

Comment:
Wrong concept modify params of ViewModel from VC. 
You should notify your VM with events such as 'onViewDidLoad'